lib/codebreaker/game.rb

```diff
@@ -0,0 +1,121 @@
+# frozen_string_literal: true
+
+require_relative 'validate_service'
+require_relative 'secret_code'
+require_relative 'version'
+require_relative 'player'
+require_relative 'data_saver'
+
+module Codebreaker
+  # Main class for codebreaker game
+  class Game
+    include(DataSaver)
+    attr_writer :secret_code
+    attr_accessor :user_code, :player, :statistic
+    attr_reader :result, :attempts, :hints, :old_hint, :difficulty_values
+
+    DIFFICULTY_VALUES = {
+      easy: { attempts: 15, hints: 2 },
+      medium: { attempts: 10, hints: 1 },
+      hell: { attempts: 5, hints: 1 }
+    }.freeze
+    def initialize
+      @user_code = []
+      load_stats
+    end
+
+    def new_game(name, difficulty)
+      generate_secret_code
+      @player = ::Codebreaker::Player.new(name)
+      difficulty_level(difficulty)
+    end
+
+    def play(user_code)
+      return nil unless ValidateService.code_valid?(user_code)
+
+      @result = []
+      check(user_code.chars, @secret_code.chars)
+      @attempts -= 1
+      save_stats if end_game?
+      result.join
+    end
+
+    def hint
+      if @hints.positive?
+        result = rand(0..3)
```

Comment:
Would it be better to set up `@hints_list = @secret_code.sample(@hints)`? And then just pop it each time.